### ISSUE TITLE
Retrieve ranking for top 10.000

### ISSUE THREAD

**Issue by TheHowl** [peppy/osu-api#102]
Perhaps also more than 10.000, although allowing it to do only at a very limited rate (like get_replay)

**Comment by wxv** [comment]
Yes, this would be useful. Otherwise I'll have to scrape the rankings. I don't imagine this being too taxing on the website. In fact, top 20k or 100k would be more useful, as top 10k doesn't cover all users of interest (ex. cheaters)

**Comment by oopsjpeg** [comment]
i don't see my self using this, but i could understand the request for this.

**Comment by firedigger** [comment]
Really need this

**Comment by grumd** [comment]
Hey @peppy! This is a very old request needed by many people here, is there any update on whether it's possible to add an endpoint? Would be enough to receive a list of IDs for up to top 100k players, or at least top 10k for each country separately (like it's done on the website). I imagine a lot of people resort to scraping the website and I really wouldn't want to put that extra load on your servers if an API was available. Thanks!

**Comment by jxu** [comment]
@grumd to actually get his attention try tweeting and mentioning him

**Comment by grumd** [comment]
So I've noticed api v2 is being implemented and this endpoint is already actually done. The only problem I have is that it seems like it's not possible to autorize for api v2 yet.

**Comment by omkelderman** [comment]
> that it seems like it's not possible to autorize for api v2 yet

It is totally possible tho, read up about laravel pasport oauth2 and look into lazer sourecode how the endpoints are called (just dont use the lazer client-id/secret, peppy has mentioned multiple times on the dev discord he doesnt want ppl to do that). Also as we speak there is work being done in osu-web to get proper scopes going

**Comment by grumd** [comment]
@omkelderman huh that's actually what I've ended up doing after spending 3 hours reading osu-web code lol. I'll look up how lazer is calling the endpoints though! That's a good hint, thanks!

Can you link the dev discord btw?

**Comment by grumd** [comment]
@omkelderman Also, you seem knowledgeable in this topic, can you help with this?
I was able to create a client using /oauth/clients, then I was able to create a token using /oauth/token with 'grant_type': 'client_credentials', but the Bearer token I received doesn't let me use any v2 endpoints, they simply return { "authentication": "basic" } even when I send an appropriate header Authorization: Bearer <token>.
I couldn't really authenticate because of this. I'll try looking at lazer code though.

**Comment by grumd** [comment]
Ok so it seems that not all types of tokens are enabled for api v2. Only grant_type of `authorization_code` and `password` seem to work. For using `password` you also need a special type of client and it seems like you can't create one yourself. Bummer, it doesn't seem like it's possible to use machine-to-machine authorization for api v2 yet.

**Comment by jxu** [comment]
@grumd try https://discordapp.com/invite/ppy

**Comment by grumd** [comment]
Yeah I already got there, thanks :D

**Comment by omkelderman** [comment]
@grumd see my comments on #227 
tldr you're suppose to create a "personal access token" but I couldnt get that to work, so I used the oauth client (which you already seem to have) set it to a localhost redirect url, went through the auth flow with my own osu account and got an access token and used that as Bearer token

**Comment by wxv** [comment]
#231 (maybe)

**Comment by jxu** [comment]
@peppy any plans to implement this?
